bgpd/bgp_evpn.c

```diff
@@ -5509,8 +5509,9 @@ int bgp_evpn_local_l3vni_add(vni_t l3vni, vrf_id_t vrf_id, struct ethaddr *rmac,
 	if (!CHECK_FLAG(bgp_vrf->vrf_flags, BGP_VRF_EXPORT_RT_CFGD))
 		evpn_auto_rt_export_add_for_vrf(bgp_vrf);
 
-	/* auto derive RD */
-	bgp_evpn_derive_auto_rd_for_vrf(bgp_vrf);
+	/* Configured RD or auto derive RD */
+	if (!CHECK_FLAG(bgp_vrf->vrf_flags, BGP_VRF_RD_CFGD))
+		bgp_evpn_derive_auto_rd_for_vrf(bgp_vrf);
```

Comment:
Doesn't bgp_evpn_derive_auto_rd_for_vrf already check this flag?

Reply:
It doesn't check for this flag.

Reply:
> How did you test this exactly?

I have configured RD under the BGP configuration of the corresponding VRF. My configuration looks like the following

vrf vrf-vni10003
 vni 10003
 exit-vrf

router bgp 200 vrf vrf-vni10003
 bgp router-id 5.1.1.2
 neighbor 55.1.1.1 remote-as 400
 neighbor 5511::1 remote-as 400
 !
 address-family ipv4 unicast
  no neighbor 5511::1 activate
 exit-address-family
 !
 address-family ipv6 unicast
  neighbor 5511::1 activate
 exit-address-family
 !
 address-family l2vpn evpn
  advertise ipv4 unicast
  advertise ipv6 unicast
   rd 3001:10003

Reply:
```/*
 * Derive RD automatically for VNI using passed information - it
 * is of the form RouterId:unique-id-for-vni.
 */
void bgp_evpn_derive_auto_rd_for_vrf(struct bgp *bgp)
{
	if (is_vrf_rd_configured(bgp))
		return;

	form_auto_rd(bgp->router_id, bgp->vrf_rd_id, &bgp->vrf_prd);
}
````

```
static inline int is_vrf_rd_configured(struct bgp *bgp_vrf)
{
	return (CHECK_FLAG(bgp_vrf->vrf_flags, BGP_VRF_RD_CFGD));
}```

Explain to me again how I am reading this wrong?

Reply:
Test :
1) Unconfigure VNI 10003
2) Configure VNI 10003

Reply:
+1 this patch looks like it has exactly the same semantics as the current code...

Reply:
I am testing on stable/7.0 that I pulled few weeks ago. I haven't noticed that this issue was already fixed in master branch. My change is not required.